core/domain/user_query_services.py

```diff
@@ -38,3 +40,30 @@ def save_new_query_model(
         query_status=feconf.USER_QUERY_STATUS_PROCESSING,
         user_ids=[]).put()
     return query_id
+
+
+def send_email_to_qualified_users(
+        query_id, email_subject, email_body, email_intent, email_option,
+        num_of_users_to_send):
+    """Send email to maximum 'num_of_users_to_send' qualified users."""
+    query_model = user_models.UserQueryModel.get(query_id)
+    query_model.query_status = feconf.USER_QUERY_STATUS_ARCHIVED
+
+    if email_option == feconf.USER_QUERY_EMAIL_OPTION_ALL:
+        bulk_email_model_id = email_manager.send_user_query_email(
+            query_model.submitter_id, query_model.user_ids, email_subject,
+            email_body, email_intent)
+        query_model.sent_email_model_id = bulk_email_model_id
+        query_model.put()
+    elif email_option == feconf.USER_QUERY_EMAIL_OPTION_CUSTOM:
+        recipient_ids = query_model.user_ids[:num_of_users_to_send]
```

Comment:
Seems like we have an issue here. According to [this](http://stackoverflow.com/questions/15377119/gae-ndb-design-performance-and-use-of-repeated-properties) we should not store more than 100-1000 values in repeated properties (my bad, sorry for not cross checking this earlier).  We will have more than 1000 values usually. So we should use JsonProperty (probably with compressed=True). We can do this change in UserQueryModel easily but this will affect BulkEmailModel where we can query on each individual user (which will not be possible in JsonProperty). We have to think of some new way to store emails sent to individual users and then change both models. What do you think  ?